### PR TITLE
Add proper concurrency support in websockets mode

### PR DESCRIPTION
**Pull request by wwwillchen**
This follows up from #1027 and it reuses the same Context instance for an entire session (i.e. websockets connection). This fixes the edge case in #1028 because the component diffing bug is fixed. See new test case which covers this.

I've extracted the environmental variables from `server/server_utils.py` into a separate package `env` to avoid cyclic dependencies since many packages like `runtime` may want to depend on `env`.

### CODE REVIEW

File: mesop/runtime/runtime.py [hunk 1]
```diff
@@ -54,8 +55,15 @@ def __init__(self):
...
     self._loading_errors: list[pb.ServerError] = []
     self._has_served_traffic = False
+    self._contexts = {}
 
   def context(self) -> Context:
+    if MESOP_WEBSOCKETS_ENABLED and hasattr(request, "sid"):
+      # flask-socketio adds sid (session id) to the request object.
+      sid = request.sid  # type: ignore
+      if sid not in self._contexts:
+        self._contexts[sid] = self.create_context()
```
Comment (richard-to) [inline mesop/runtime/runtime.py:65]: [question] Is runtime created every request or is it a singleton? If it's the latter, I was wondering if _contexts would need to be cleared

File: mesop/tests/e2e/concurrent_updates_websockets_test.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {testInWebSocketsEnabledOnly} from './e2e_helpers';
+import {expect} from '@playwright/test';
+
+testInWebSocketsEnabledOnly(
```
Comment (richard-to) [inline mesop/tests/e2e/concurrent_updates_websockets_test.ts:4]: Any other test cases we should check that wouldn't be covered by existing tests?

**Review by richard-to — APPROVED** [review]
Looks good. Think I only had one question I was unsure of in the code.
